### DISCUSSION TITLE
Set Field Value(s) (Patch)

### DISCUSSION THREAD

**Discussion by fourgates** [TanStack/form#649] [Q&A]
Angular forms have a `patchValue` [api](https://angular.io/guide/reactive-forms#updating-parts-of-the-data-model). This allows you to to update `n` number of fields in the form. 

```
  this.profileForm.patchValue({
    firstName: 'Nancy',
    address: {
      street: '123 Drew Street',
    },
  });
```

This is useful when you are editing a form, loading an entity from the server, and want to set all the values (patch). Looking at the current API, this would be a manual process.

```
this.form.setFieldValue('firstName', 'tony');
this.form.setFieldValue('lastName', 'stanza');
```

FWIW -- #613 is kind of related. I'm not sure if `update` is supposed to accomplish this task? Is there a vision of how to accomplish this with tanstack?

**Comment by mistyharsh** [comment]
+1. I have the same question! Is there any official solution for this?

**Comment by 0x-DLN** [comment]
+1, it's really annoying to `setFieldValue(...)` manually like 20 times when editing a big form.

You can set them on the form object using `form.state.value` but it doesn't trigger a validation & UI update, and I can't find a method to do that

**Reply by 2Bros-MX** [reply]
Hi, have u found a way to trigger the validation and UI update? I'm having the same issue after using setFieldValue(...). It doesnt trigger my async validator for onChange.

**Comment by LeCarbonator** [comment]
With some restructuring, you can pass an updater function to `'namespace'` where your fields are.

So instead of 
```ts
this.form.setFieldValue('firstName', 'tony');
this.form.setFieldValue('lastName', 'stanza');
```

You'd have
```ts
this.form.setFieldValue('person', (previous) => ({...previous, firstName: 'tony', lastName: 'stanza'}))
```
